src/TLuaInterpreter.cpp

```diff
@@ -10739,6 +10750,505 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+            lua_pushboolean(L, false);
+            return 1;
+        }
+        
+        lua_pushstring(L, speechQueue.at(index).toLatin1().constData());
+        return 1;
+    }
+    
+    lua_newtable(L);
+    
+    for(int i=0;i<speechQueue.size();i++) {
+        lua_pushnumber(L, i + 1);
+        lua_pushstring(L, speechQueue.at(i).toUtf8().constData());
+        lua_settable(L, -3);
+    }
+    
+    return 1;
+}
+
+/** ttsPauseSpeech( opt. pause )
+ *  Pauses speech if passed true, resumes if passed false.
+ *  If no input, returns whether or not the queue is paused.
+ */
+int TLuaInterpreter::ttsPauseSpeech(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    if(!lua_isboolean(L, 1)) {
+        lua_pushfstring(L, "ttsPauseSpeech: bad argument #1 type (pause expected bool, instead got %s!)", luaL_typename(L, 1));
+        lua_error(L);
+        return 1;
+    }
+    
+    if(lua_toboolean(L, 1)) {
+        speechUnit->pause();
+    } else {
+        speechUnit->resume();
+    }
+	
+    return 0;
+}
+
+/** ttsClearQueue( opt. index )
+ *  Removes an item from the speech queue at index.
+ *  Returns false if invalid index provided.
+ *    Clears entire queue if 
+ */
+int TLuaInterpreter::ttsClearQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    if(lua_gettop(L) > 0) {
+        if(!lua_isnumber(L, 1)) {
+            lua_pushfstring(L, "ttsClearQueue: bad argument #1 type (optional index as int expected, got %s!)", luaL_typename(L, 1));
+            lua_error(L);
+            return 1;
+        }
+        
+        int index = lua_tonumber(L, 1);
+        index--;
+        
+        if(index < 0 || index >= speechQueue.size()) {
+            lua_pushfstring(L, "ttsClearQueue: index (%d) out of bounds for queue size %d", index+1, speechQueue.size());
+            lua_error(L);
+            return 1;
+        }
+        
+        speechQueue.remove(index);
+        return 0;
+    }
+    
+    speechQueue.clear();
+    return 0;
+}
+
+/** ttsGetCurrentLine()
+ *  Returns the text being spoken, or false if not speaking.
+ */
+int TLuaInterpreter::ttsGetCurrentLine(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    if(speechUnit->state() == QTextToSpeech::Ready) {
+        lua_pushboolean(L, false);
+        return 1;
+    }
+    
+    lua_pushstring(L, speechCurrent.toUtf8().constData());
+    return 1;
+}
+
+/** ttsGetState()
+ *  Returns the current state of the speechUnit.
+ *  Possible returns: speaking, ready, paused, error, unknown.
+ */
+int TLuaInterpreter::ttsGetState(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    switch(speechUnit->state())
+    {
+        case QTextToSpeech::Ready : lua_pushstring(L, "ready");
+        case QTextToSpeech::Paused : lua_pushstring(L, "paused");
+        case QTextToSpeech::Speaking : lua_pushstring(L, "speaking");
+        case QTextToSpeech::BackendError : lua_pushstring(L, "error");
+        default : lua_pushstring(L, "unknown");
```

Comment:
These strings differ from the ones used in the events above. We should probably use the same in both cases.